sdk/identity/azure-identity/src/main/java/com/azure/identity/InteractiveBrowserCredentialBuilder.java

```diff
@@ -89,9 +109,8 @@ InteractiveBrowserCredentialBuilder disableAutomaticAuthentication() {
     public InteractiveBrowserCredential build() {
         ValidationUtil.validate(getClass().getSimpleName(), new HashMap<String, Object>() {{
                 put("clientId", clientId);
-                put("port", port);
```

Comment:
Should either `port` or `redirectUrl` be required?

Reply:
nope, 
if both are not specified, then we default to localhost and MSAL will find an open port to connect to.

Reply:
> MSAL will find an open port to connect to

MSAL doesn't connect to this redirectUrl. AAD will verify it matches the one in the app registration and only if it completely matches, will then `login.microsoftonline.com` send the user to this URL, in the browser.

Reply:
So, the redirect URL is only required, if user is trying to authenticate with their custom registered application.
If they are using a first party client id, then the redirectURL can be arbitrary in the sense that Localhost can be used with any port and user doesn't need to supply it.